src/main/java/com/epam/reportportal/service/tree/ItemTreeReporter.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2019 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.reportportal.service.tree;
+
+import com.epam.reportportal.message.TypeAwareByteSource;
+import com.epam.reportportal.restendpoint.http.MultiPartRequest;
+import com.epam.reportportal.service.ReportPortalClient;
+import com.epam.reportportal.utils.MimeTypeDetector;
+import com.epam.reportportal.utils.http.HttpRequestUtils;
+import com.epam.ta.reportportal.ws.model.*;
+import com.epam.ta.reportportal.ws.model.item.ItemCreatedRS;
+import com.epam.ta.reportportal.ws.model.log.SaveLogRQ;
+import com.google.common.collect.Lists;
+import com.google.common.io.Files;
+import io.reactivex.Maybe;
+import io.reactivex.MaybeSource;
+import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Function;
+import io.reactivex.schedulers.Schedulers;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Date;
+
+/**
+ * This class provides methods for sending requests to the Report Portal instance, using {@link ReportPortalClient}
+ * and {@link com.epam.reportportal.service.tree.TestItemTree.TestItemLeaf}.
+ * Provided requests:
+ * - start test item
+ * - finish test item
+ * - send log
+ *
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public class ItemTreeReporter {
```

Comment:
I have no idea how I suppose to use this class on practice. Could you provide any examples?

Reply:
repository wiki updated with description